motionplan/utils.go

```diff
@@ -43,6 +43,22 @@ func (plan Plan) String() string {
 	return str
 }
 
+// Evaluate assigns a numeric score to a plan that corresponds to the cumulative distance between input waypoints in the plan.
+func (plan Plan) Evaluate(distFunc ik.SegmentMetric) (totalCost float64) {
+	if len(plan) < 2 {
+		return math.Inf(1)
+	}
+	for i := 0; i < len(plan)-1; i++ {
+		for component, inputs := range plan[i] {
+			if nextInputs, ok := plan[i+1][component]; ok && len(inputs) > 0 {
```

Comment:
1. Yes it is. A plan can say "move this component from here, to here, for this step" and then not mention it again.
2. There shouldn't be such a thing as an invalid plan. However, since there's no need for *adjacency* of input calls to a given component, I should be comparing against "last inputs to this component" rather than last step. Will update.
3. All 0-dof components will have length 0, and are allowed to be included in the plan.